app/scripts/metamask-controller.js

```diff
@@ -409,6 +420,10 @@ export default class MetamaskController extends EventEmitter {
         networkControllerMessenger,
         'NetworkController:infuraIsUnblocked',
       ),
+      onAccountRemoved: keyringControllerMessenger.subscribe.bind(
```

Comment:
We should be using the top-level controller messenger for subscriptions in this file, e.g.

```suggestion
      onAccountRemoved: this.controllerMessenger.subscribe.bind(
```

This will allow us to move the keyring controller messenger constructor back down to where it was originally as well